src/main/java/de/zalando/aruha/nakadi/validation/RFC3339DateTimeValidator.java

```diff
@@ -0,0 +1,39 @@
+package de.zalando.aruha.nakadi.validation;
+
+import org.everit.json.schema.FormatValidator;
+
+import java.time.format.DateTimeParseException;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static java.time.format.DateTimeFormatter.ISO_OFFSET_DATE_TIME;
+
+public class RFC3339DateTimeValidator implements FormatValidator {
+
+    private final String errorMessage = "must be a valid date-time";
+
+    // Valid offsets are either Z or hh:mm. The format hh:mm:ss is not valid
+    private final String dateTimeOffsetPattern = "^.*(Z|((\\+|-)\\d\\d:\\d\\d))$";
```

Comment:
@rcillo 
Yep, unfortunately this solution does not check the format but only ignores seconds.
If your main purpose is to reject second precision offsets then you can do this:
```java
       OffsetDateTime d = OffsetDateTime.parse("1996-10-15T16:39:57+07:00:30", DateTimeFormatter.ISO_OFFSET_DATE_TIME);
        if (d.get(ChronoField.OFFSET_SECONDS) % 60 != 0) {
            throw new IllegalArgumentException("Offset with seconds not supported");
        }
```